arviz/utils.py

```diff
@@ -320,3 +323,167 @@ def expand_dims(x):
 def full(shape, x, dtype=None):
     """Jitting numpy full."""
     return np.full(shape, x, dtype=dtype)
+
+
+def flat_inference_data_to_dict(
+    data,
+    var_names=None,
+    groups=None,
+    dimensions=None,
+    group_info=False,
+    var_name_format=None,
+    index_origin=None,
+):
+    """Transform data to dictionary.
+
+    Parameters
+    ----------
+    data : obj
+        Any object that can be converted to an az.InferenceData object
+        Refer to documentation of az.convert_to_inference_data for details
+    var_names : str or list of str, optional
+        Variables to be processed, if None all variables are processed.
+    groups : str or list of str, optional
+        Select groups for CDS. Default groups are {"posterior_groups", "prior_groups"}
+            - posterior_groups: posterior, posterior_predictive, sample_stats
+            - prior_groups: prior, posterior_predictive, sample_stats_prior
+    ignore_groups : str or list of str, optional
+        Ignore specific groups from CDS.
+    dimension : str, or list of str, optional
+        Select dimensions along to slice the data. By default uses ("chain", "draw").
+    group_info : bool
+        Add group info for `var_name_format`
+    var_name_format : str or tuple of tuple of string, optional
+        Select column name format for non-scalar input.
+        Predefined options are {"brackets", "underscore", "cds"}
+            "brackets":
+                - add_group_info == False: theta[0,0]
+                - add_group_info == True: theta_posterior[0,0]
+            "underscore":
+                - add_group_info == False: theta_0_0
+                - add_group_info == True: theta_posterior_0_0_
+            "cds":
+                - add_group_info == False: theta_ARVIZ_CDS_SELECTION_0_0
+                - add_group_info == True: theta_ARVIZ_GROUP_posterior__ARVIZ_CDS_SELECTION_0_0
+            tuple:
+                Structure:
+                    tuple: (dim_info, group_info)
+                        dim_info: (str: `.join` separator,
+                                   str: dim_separator_start,
+                                   str: dim_separator_end)
+                        group_info: (str: group separator start, str: group separator end)
+                Example: ((",", "[", "]"), ("_", ""))
+                    - add_group_info == False: theta[0,0]
+                    - add_group_info == True: theta_posterior[0,0]
+    index_origin : int, optional
+        Start parameter indeces from `index_origin`. Either 0 or 1.
```

Comment:
I think it would be better to only use this when creating the inference data. From what I understand it is happening here (I have had no time to try it, only read the code) this assumes the index origin in InferenceData is always 0 and it adds one if index origin is 1. However, there is the possibility of the InferenceData index origin being 2000 if `.sel` has been used to discard some burn-in, so the index origin 1 would change the start from 2000 to 2001.

Reply:
Yeah, this makes sense

Reply:
No, I think index origin is only an artifact for flattened names. InferenceData is always 0-based.

Reply:
But I need to make syre that selection works as intended

Reply:
I have been thinking a little bit more about indexing, and it does make sense to only use it for visualization, not in Inference Data. We should probably create one issue with what is missing to integrate the rcparam in ArviZ.

The function in plot utils to make labels should probably modified, summary should also use the rcparam value as default, and maybe `sel` should also have an index origin argument? Otherwise the index shown in the plots cannot be used for selection right away. Would this be an issue?

Reply:
Yeah, I think those are currently only locations which needs a change.

Reply:
I changed this so it currently uses numpy indices, so it should work in all the cases for this function.

I also added rcparam for summary.

.sel thing needs some thinking.